src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -360,6 +368,16 @@ def load_arguments(self, _):
     with self.argument_context('image builder validator add', min_api='2022-02-14') as c:
         c.argument('dis_on_failure', options_list=['--continue-distribute-on-failure', '--dis-on-failure'], arg_type=get_three_state_flag(), help="If validation fails and this parameter is set to false, output image(s) will not be distributed.")
         c.argument('source_validation_only', arg_type=get_three_state_flag(), help="If this parameter is set to true, the image specified in the 'source' section will directly be validated. No separate build will be run to generate and then validate a customized image.")
+
+    with self.argument_context('image builder optimizer add', min_api='2022-07-01') as c:
+        c.argument('optimizer_type', get_enum_type(['vmBoot']), options_list=['--type', '--optimizer-type'], help='The type of optimizer added.')
```

Comment:
Is this enumeration only has one value? Will there be more types of `--optimizer-type` in the future?

Reply:
Currently there is only one type of `--optimizer-type`. 
https://github.com/Azure/azure-cli/issues/25853#issue-1628367982

Reply:
In fact, in the Swagger, the `vmBoot` property is bool type, I personally think that our parameter design should be consistent with Swagger.
https://github.com/Azure/azure-rest-api-specs/blob/9f1fc4ec1d4246f5f1f594541fe76fdd8a0712a8/specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-07-01/imagebuilder.json#L1507-L1525

And if a parameter can only support passing in one value, making it a Boolean type would be more concise.

Reply:
May I ask what is the reason for designing `vmBoot` as a `--type {vmBoot} --state {enabled:disabled}` parameters? Will the `--type` parameter expand to more values in the future?

Reply:
@zdelacerda-microsoft I have a question to confirm. If `--type` expands to other values in the future, can these values be used simultaneously or can only one of them be used?

Reply:
> Prefetch (3 new commands, also a new "optimize" sub category similar to customizer functionality)
> 
> 1. az image builder optimizer add --optimizer-name --type {vmBoot} --state {enabled:disabled} add optimizer to an image template
> 2. az image builder optimizer clear removes all optimizers
> 3. az image builder optimizer remove --optimizer-name removes an optimizer

@zdelacerda-microsoft 
since the usual practice in CLI is using `update` command to change the status of a resource property and `remove` command to remove the entile resource, it is suggested to change the format of this group to the following format.
Could you please help to check if it could meet your expectation?
1. `az image builder optimizer add --resource-group-name MyResourceGroup --image-template-name MyTemplate --enable-vm-boot {true:false}` add optimizer to an image template
2. `az image builder optimizer update --resource-group-name MyResourceGroup --image-template-name MyTemplate --enable-vm-boot {true:false}` update optimizer of an image template
3. `az image builder optimizer remove --resource-group-name MyResourceGroup --image-template-name MyTemplate` remove optimizers from an image template

BTW, this format has not been determined yet and there is a code freeze this Friday, the change of this group has to be postponed to next sprint (7-4), `--validator` parameter in `az image builder create` as well (https://github.com/Azure/azure-cli/issues/25853#issuecomment-1543267336). 
The rest part of this feature request could be released in this sprint.

Reply:
@yanzhudd, examples you mentioned are fine, we want to go with --enable-vm-boot, but state you mentioned is boolean, can it be {enabled:disabled} or {true:false} will will mapped to it? any way will be fine

Reply:
@zhoxing-ms, --type may not be expand in future, but if so it will be used simultaneously not one at a time

Reply:
This looks good to me. Thank you @yanzhudd and @zhoxing-ms